packages/mui-base/src/utils/useScrollLock.ts

```diff
@@ -1,75 +1,110 @@
-import { useEnhancedEffect } from './useEnhancedEffect';
-import { useId } from './useId';
 import { isIOS } from './detectBrowser';
+import { useEnhancedEffect } from './useEnhancedEffect';
+
+let originalRootStyles = {};
+let originalBodyStyles = {};
+let preventScrollCount = 0;
+let restore: () => void = () => {};
+
+function preventScrollIOS() {
+  // To implement
+  return () => {};
+}
+
+function preventScrollStandard() {
+  const html = document.documentElement;
+  const body = document.body;
+  const rootStyle = html.style;
+  const bodyStyle = body.style;
```

Comment:
We need to read the computed styles, otherwise the checks below work only when html/body styles are inline